packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/hooks.ts

```diff
@@ -0,0 +1,78 @@
+import { RefObject, useEffect, useState } from 'react';
+
+import { TimeOption } from '@grafana/data';
+
+const modulo = (a: number, n: number) => ((a % n) + n) % n;
+
+/** @internal */
+export interface UseListFocusProps {
+  localRef: RefObject<HTMLUListElement>;
+  options: TimeOption[];
+}
+
+/** @internal */
+export type UseListFocusReturn = [(event: React.KeyboardEvent) => void];
+
+/** @internal */
+export const useListFocus = ({ localRef, options }: UseListFocusProps): UseListFocusReturn => {
+  const [focusedItem, setFocusedItem] = useState(0);
+
+  useEffect(() => {
+    // Reset focused item when options have changed
+    setFocusedItem(0);
```

Comment:
If an item is selected, that's the one supposed to get the focus if I remember correctly.

Reply:
Hmm good point, that means somehow the list will have to have a selected concept or receive it

Reply:
You could query the localRef for the `checked` attribute and then see which index it is in the list.

Reply:
I have done something like this, but open to ideas on how to rewrite it in a more clear way if there is one 👍